portfolio/src/main/webapp/script.js

```diff
@@ -575,12 +574,19 @@ function sendFormData(blogNumber, commentForm, imageUploadUrl) {
   const req = new XMLHttpRequest();
   req.open("POST", imageUploadUrl, true);
   req.onload = function() {
-    if (req.status == 200) {
-      console.log("Uploaded!");
-
-      console.log("Reloading comments and chart");
-      loadCommentSection(blogNumber);
+    // when request finished, response is ready, and status is ok
```

Comment:
onload implies readyState = 4. You just have to check the status and response.

Also, add an onerror.